torchnlp/datasets/glue.py

```diff
@@ -0,0 +1,231 @@
+            nb_arguments = len(keys)
+            for line in tsvreader:
+                examples.append({keys[i]: line[i] for i in range(nb_arguments)})
+        ret.append(Dataset(examples))
+    elif dataset == 'CoLA':
+        for filename in splits:
+            examples = []
+            if filename == 'test.tsv':
+                with open(os.path.join(folder_path, filename), newline='') as tsvfile:
+                    tsvreader = csv.reader(tsvfile, delimiter='\t', quoting=csv.QUOTE_NONE)
+                    keys = next(tsvreader)
+                    nb_arguments = len(keys)
+                    for line in tsvreader:
+                        examples.append({keys[i]: line[i] for i in range(nb_arguments)})
+            else:
+                with open(os.path.join(folder_path, filename), newline='') as tsvfile:
+                    tsvreader = csv.reader(tsvfile, delimiter='\t', quoting=csv.QUOTE_NONE)
+                    keys = ['source', 'acceptability judgment', 'original acceptability judgment', 'sentence']
+                    nb_arguments = len(keys)
+                    for line in tsvreader:
+                        examples.append({keys[i]: line[i] for i in range(nb_arguments)})
+            ret.append(Dataset(examples))
+    elif dataset == 'MRPC':
+        for filename in splits:
+            examples = []
+            with open(os.path.join(folder_path, filename), newline='') as tsvfile:
+                tsvreader = csv.reader(tsvfile, delimiter='\t', quoting=csv.QUOTE_NONE)
+                keys = next(tsvreader)
+                if filename != 'test.tsv':
+                    # This had to be done because of an unexplained string '\ufeff' appearing sometimes in front of 'Quality' during mrpc_processing() 
+                    keys = ['Quality', '#1 ID', '#2 ID', '#1 String', '#2 String'] 
+                nb_arguments = len(keys)
+                for line in tsvreader:
+                    examples.append({keys[i]: line[i] for i in range(nb_arguments)})
+            ret.append(Dataset(examples))
+    else:
+        for filename in splits:
+            examples = []
+            with open(os.path.join(folder_path, filename), newline='') as tsvfile:
+                tsvreader = csv.reader(tsvfile, delimiter='\t', quoting=csv.QUOTE_NONE)
+                keys = next(tsvreader)
+                nb_arguments = len(keys)
+                for line in tsvreader:
+                    """
+                        problem with QQP and SNLI : len(line) is different from nb_arguments for some lines ...
+                    """
+                    if len(line) == nb_arguments:
+                        examples.append({keys[i]: line[i] for i in range(nb_arguments)})
+                    else:
+                        pass
+            ret.append(Dataset(examples))
+
+    if len(ret) == 1:
+        return ret[0]
+    else:
+        return tuple(ret)
+
+
+def mrpc_processing(folder_path, mrpc_train_file, mrpc_test_file):
```

Comment:
There is no comments on this function, please document it. It's hard to understand what it's doing.